youtrack_sdk/entities.py

```diff
@@ -327,3 +327,35 @@ class IssueLink(BaseModel):
     link_type: Optional[IssueLinkType] = Field(alias="linkType")
     issues: Optional[Sequence[Issue]]
     trimmed_issues: Optional[Sequence[Issue]] = Field(alias="trimmedIssues")
+
+
+class AgileInfo(BaseModel):
+    type: Literal["Agile"] = Field(alias="$type", default="Agile")
+    id: Optional[str]
+    name: Optional[str]
+
+
+class SprintInfo(BaseModel):
+    type: Literal["Sprint"] = Field(alias="$type", default="Sprint")
+    id: Optional[str]
+    name: Optional[str]
+    goal: Optional[str]
+    start: Optional[datetime]
+    finish: Optional[datetime]
+    archived: Optional[bool]
+    is_default: Optional[bool] = Field(alias="isDefault")
+    unresolvedIssuesCount: Optional[int]
```

Comment:
I'm a bit confused, shouldn't those be part of `Sprint` instead of `SprintInfo` according to your system?

Reply:
It depends on the sprint information we want to get from the linked sprints in the `Agile` object. I have one suggestion:

1. Provide only `type`, `id` and `name` fields in the `AgileBase` and `SprintBase` classes.
2. Rename `AgileBase` and `SprintBase` to something like `AgileLink` / `AgileRef` / `AgileReference`

Thus, `get_agiles` / `get_sprints` will only return the ids and names of the linked sprints/agiles. And if you need full information about the corresponding agile or sprint, you can use `get_agile` / `get_sprint` methods.

What do you think?

Reply:
I like your idea, I think this makes it consistent. In terms of naming, I would go for `AgileRef` and `SprintRef` - this sounds most natural to me.

Reply:
Done.